frontend/document/document.lua

```diff
@@ -93,9 +93,19 @@ end
 -- this might be overridden by a document implementation
 function Document:close()
     local DocumentRegistry = require("document/documentregistry")
-    if self.is_open and DocumentRegistry:closeDocument(self.file) == 0 then
-        self.is_open = false
-        self._document:close()
+    if self.is_open then
+        if DocumentRegistry:closeDocument(self.file) == 0 then
+            self.is_open = false
+            self._document:close()
+            self._document = nil
+
+            -- NOTE: DocumentRegistry:openDocument will force a GC sweep the next time we open a Document.
+            --       MµPDF will also do a bit of spring cleaning of its internal cache when opening a *different* document.
+        else
+           logger.warn("Tried to close a document with *multiple* remaining hot references")
```

Comment:
(3 spaces indentation)

Reply:
Crap. That happens a lot with Kate and Lua for some reason.